aiida/cmdline/commands/group.py

```diff
@@ -29,556 +34,281 @@ def __init__(self):
+        sys.exit(0)
 
-        if created:
-            print "Group created with PK = {} and name '{}'".format(
-                group.pk, group.name)
-        else:
-            print "Group '{}' already exists, PK = {}".format(
-                group.name, group.pk)
+    group.remove_nodes(nodes)
 
-    def group_rename(self, *args):
-        """
-        Rename an existing group.
-        """
-        if not is_dbenv_loaded():
-            load_dbenv()
 
-        import argparse
-        from aiida.orm import Group as G
-        from aiida.orm import load_group
+@verdi_group.command("addnodes")
+@options.GROUP()
+@arguments.NODES()
+@with_dbenv()
+def group_addnodes(group, nodes, *args):
+    """
+    Add NODES to a given AiiDA group.
+    """
+    from aiida.cmdline import wait_for_confirmation
 
-        parser = argparse.ArgumentParser(
-            prog=self.get_full_command_name(),
-            description='Rename an existing group.')
-        parser.add_argument('GROUP', help='The name or PK of the group to rename')
-        parser.add_argument('NAME', help='The new name of the group')
+    group_pk = group.pk
+    group_name = group.name
 
-        args = list(args)
-        parsed_args = parser.parse_args(args)
+    echo.echo("Are you sure to add {} nodes the group with PK = {} "
+            "({})? [Y/N] ".format(len(nodes), group_pk, group_name))
+    if not wait_for_confirmation():
+        sys.exit(0)
 
-        group = parsed_args.GROUP
-        name = parsed_args.NAME
+    group.add_nodes(nodes)
 
-        try:
-            group_pk = int(group)
-        except ValueError:
-            group_pk = None
-            group_name = group
-
-        if group_pk is not None:
-            try:
-                group = G(dbgroup=group_pk)
-            except NotExistent as e:
-                print >> sys.stderr, "Error: {}.".format(e.message)
-                sys.exit(1)
-        else:
-            try:
-                group = G.get_from_string(group_name)
-            except NotExistent as e:
-                print >> sys.stderr, "Error: {}.".format(e.message)
-                sys.exit(1)
+@verdi_group.command("delete")
+@arguments.GROUP()
+@options.FORCE(help="Force deletion of the group even if it "
+                    "is not empty. Note that this deletes only the "
+                    "group and not the nodes.")
+@with_dbenv()
+def group_delete(group, force, *args):
+    """
+    Pass the GROUP to delete an existing group.
+    """
+    from aiida.cmdline import wait_for_confirmation
+
+    group_pk = group.pk
+    group_name = group.name
+
+    num_nodes = len(group.nodes)
+    if num_nodes > 0 and not force:
+        echo.echo_critical(("Group '{}' is not empty (it contains {} "
+                              "nodes). Pass the -f option if you really want to delete "
+                              "it.".format(group_name, num_nodes)))
+
+    echo.echo("Are you sure to kill the group with PK = {} ({})? "
```

Comment:
For all the "wait_for_confirmation" occurrences, it is better to use the `click` built in.
For example from `verdi calculation kill`:
```
    if not force:
        click.confirm('Are you sure you want to kill {} calculations?'.format(len(calculations)), abort=True)
```

This will prompt, unless `force` is set, and if the prompt is not responded with yes, the command is aborted, which is what you are trying to achieve here